ivy.el

```diff
@@ -3315,26 +3315,35 @@ FACE is the face to apply to STR."
                   (string-match "^[^:]+:[^:]+:" str))
              (match-end 0)
            0))
-        (re (ivy-generic-regex-to-str ivy--old-re)))
-    (ignore-errors
-      (while (and (string-match re str start)
-                  (> (- (match-end 0) (match-beginning 0)) 0))
-        (setq start (match-end 0))
-        (let ((i 0))
-          (while (<= i ivy--subexps)
-            (let ((face
-                   (cond ((zerop ivy--subexps)
-                          (cadr ivy-minibuffer-faces))
-                         ((zerop i)
-                          (car ivy-minibuffer-faces))
-                         (t
-                          (nth (1+ (mod (+ i 2)
-                                        (1- (length ivy-minibuffer-faces))))
-                               ivy-minibuffer-faces)))))
-              (ivy-add-face-text-property
-               (match-beginning i) (match-end i)
-               face str))
-            (cl-incf i))))))
+        (regexps
+         (if (listp ivy--old-re)
+             (mapcar
+              #'car
+              (cl-remove-if
+               (lambda (regexp-cons)
+                 (null (cdr regexp-cons)))
+               ivy--old-re))
```

Comment:
Why not

```el
(cl-mapcan (lambda (cell)
             (and (cdr cell)
                  (list (car cell))))
           ivy--old-re)
```

or, at the very least,

```el
(mapcar #'car (cl-remove-if-not #'cdr ivy--old-re))
```

(guaranteed 100% untested, mind you)